packages/crypto/src/crypto.ts

```diff
@@ -0,0 +1,498 @@
+  return uint8arrayFromHexString(uncompressedHexString);
+};
+/**
+ * Compute the modular square root using the Tonelli-Shanks algorithm.
+ */
+
+const modSqrt = (x: bigint, p: bigint): bigint => {
+  if (p <= BigInt(0)) {
+    throw new Error("p must be positive");
+  }
+  const base = x % p;
+
+  // Check if p % 4 == 3 (applies to NIST curves P-256, P-384, and P-521)
+  if (testBit(p, 0) && testBit(p, 1)) {
+    const q = (p + BigInt(1)) >> BigInt(2);
+    const squareRoot = modPow(base, q, p);
+    if ((squareRoot * squareRoot) % p !== base) {
+      throw new Error("could not find a modular square root");
+    }
+    return squareRoot;
+  }
+
+  // Other elliptic curve types not supported
+  throw new Error("unsupported modulus value");
+};
+
+/**
+ * Compute the modular exponentiation.
+ */
+const modPow = (b: bigint, exp: bigint, p: bigint): bigint => {
+  if (exp === BigInt(0)) {
+    return BigInt(1);
+  }
+  let result = b;
+  const exponentBitString = exp.toString(2);
+  for (let i = 1; i < exponentBitString.length; ++i) {
+    result = (result * result) % p;
+    if (exponentBitString[i] === "1") {
+      result = (result * b) % p;
+    }
+  }
+  return result;
+};
+
+/**
+ * Test if a specific bit is set.
+ */
+const testBit = (n: bigint, i: number): boolean => {
+  const m = BigInt(1) << BigInt(i);
+  return (n & m) !== BigInt(0);
+};
+
+/**
+ * Generate a random Uint8Array of a specific length.
+ */
+const randomBytes = (length: number): Uint8Array => {
+  const array = new Uint8Array(length);
+  for (let i = 0; i < length; i++) {
+    array[i] = Math.floor(Math.random() * 256);
+  }
+  return array;
+};
+
+/**
+ * Create additional associated data (AAD) for AES-GCM decryption.
+ */
+const additionalAssociatedData = (
+  senderPubBuf: ArrayBuffer,
+  receiverPubBuf: ArrayBuffer
+): Uint8Array => {
+  const s = Array.from(new Uint8Array(senderPubBuf));
+  const r = Array.from(new Uint8Array(receiverPubBuf));
+  return new Uint8Array([...s, ...r]);
+};
+
+/**
+ * Perform HKDF extract and expand operations.
+ */
+const extractAndExpand = async (
+  sharedSecret: Uint8Array,
+  ikm: Uint8Array,
+  info: Uint8Array,
+  len: number
+): Promise<Uint8Array> => {
+  const prk = hkdf.extract(sha256, ikm, sharedSecret);
+  const resp = hkdf.expand(sha256, prk, info, len);
+  return new Uint8Array(resp);
+};
+
+/**
+ * Derive the Diffie-Hellman (DH) shared secret using ECDH.
+ */
+const deriveDh = async (
+  encappedKeyBuf: Uint8Array,
+  receiverPriv: string
+): Promise<Uint8Array> => {
+  const dh = p256.getSharedSecret(
+    base64urlDecode(receiverPriv),
+    encappedKeyBuf
+  );
+  return dh.slice(1);
+};
+
+/**
+ * Decrypt data using AES-GCM.
+ */
+const aesGcmDecrypt = (
+  encryptedData: Uint8Array,
+  key: Uint8Array,
+  iv: Uint8Array,
+  aad?: Uint8Array
+): Uint8Array => {
+  const aes = gcm(key, iv, aad);
+  const data_ = aes.decrypt(encryptedData);
+  return data_;
```

Comment:
drop the underscore (unless there's a good reason for it). Or even better, get rid of this variable entirely: `return aes.decrypt(encryptedData)`

Reply:
good call - copy pasta from an example haha